### PR TITLE
api: rest: handle null test_name in project/test_results

### PR DESCRIPTION
**Pull request by chaws**
Fixes https://github.com/Linaro/squad/issues/729

### CODE REVIEW

File: squad/api/rest.py [hunk 1]
```diff
@@ -347,9 +348,16 @@ class Meta:
 
 class LatestTestResults(object):
     def __init__(self, build, test_name):
+        if test_name is None:
+            raise serializers.ValidationError(_('"test_name" parameter is mandatory. Ex: suitename/testname'))
+
+        suite_and_test = test_name.rsplit("/", 1)
+        if len(suite_and_test) != 2:
```
Comment (mwasilew) [inline squad/api/rest.py:355]: I'm afraid this might break CTS results. Could you check this patch with any CTS job?

Comment (chaws) [reply squad/api/rest.py:355]: Humm, good point. I have tested with the following url: `http://localhost:8000/api/projects/116/test_results/?test_name=cts-lkft-arm64-v8a/arm64-v8a.CtsOsTestCases/android.os.cts.BuildTest.testSdkInt`

and this is a piece of the json returned:

```json
{
        "build": {
            "url": "http://localhost:8000/api/builds/10138/",
            "id": 10138,
            "testruns": "http://localhost:8000/api/builds/10138/testruns/",
            "testjobs": "http://localhost:8000/api/builds/10138/testjobs/",
            "status": "http://localhost:8000/api/builds/10138/status/",
            "metadata": "http://localhost:8000/api/builds/10138/metadata/",
            "finished": true,
            "version": "980",
            "created_at": "2018-10-07T23:46:06.716141Z",
            "datetime": "2018-10-07T23:46:06.715963Z",
            "patch_id": null,
            "keep_data": false,
            "project": "http://localhost:8000/api/projects/116/",
            "patch_source": null,
            "patch_baseline": null
        },
        "build_url_path": "/android-lkft/aosp-master-tracking/build/980/",
        "environments": [
            {
                "environment": {
                    "url": "http://localhost:8000/api/environments/352/",
                    "id": 352,
                    "slug": "hi6220-hikey_4.14",
                    "name": null,
                    "expected_test_runs": null,
                    "description": null,
                    "project": "http://localhost:8000/api/projects/116/"
                },
                "test": {
                    "id": 88718197,
                    "name": "cts-lkft-arm64-v8a/arm64-v8a.CtsOsTestCases/android.os.cts.BuildTest.testSdkInt",
                    "short_name": "android.os.cts.BuildTest.testSdkInt",
                    "status": "fail",
                    "result": false,
                    "log": null,
                    "has_known_issues": false,
                    "test_run": 960358,
                    "suite": 4226,
                    "known_issues": []
                },
                "test_url_path": "/android-lkft/aosp-master-tracking/tests/cts-lkft-arm64-v8a/arm64-v8a.CtsOsTestCases/android.os.cts.BuildTest.testSdkInt"
            },
...
```

Comment (chaws) [reply squad/api/rest.py:355]: I guess it's working fine

Comment (mwasilew) [reply squad/api/rest.py:355]: looks like it.

**Pull request merged**